Public/Src/Engine/UnitTests/Scheduler.IntegrationTest/ChangeAffectedInputTests.cs

```diff
@@ -74,6 +74,93 @@ public void DirectAffectedFileInputTest()
             XAssert.AreEqual(expectedAffectedInput, actualAffectedInput);
         }
 
+        [Fact]
+        public void RevertChangeCacheMissTest()
+        {
+            // aInput-->(pipA)-->aOutput-->(pipC)-->cOutput
+            //                            /                         
+            // bInput-->(pipB)-->bOutput--
+
+            
+            var dir = Path.Combine(ObjectRoot, "Dir");
+            var dirPath = AbsolutePath.Create(Context.PathTable, dir);
+
+            FileArtifact changeList = CreateSourceFile(root: dirPath, prefix: "changeList-file");
+
+            // Process A.
+            FileArtifact aInput = CreateSourceFile(root: dirPath, prefix: "pip-a-input-file");
+            FileArtifact aOutput = CreateOutputFileArtifact(root: dirPath, prefix: "pip-a-out-file");
+            File.WriteAllText(aInput.Path.ToString(Context.PathTable), "pipA");
+
+            var pipBuilderA = CreatePipBuilder(new[] { Operation.ReadFile(aInput), Operation.WriteFile(aOutput) });
+            var pipA = SchedulePipBuilder(pipBuilderA);
+
+            // Process B.
+            FileArtifact bInput = CreateSourceFile(root: dirPath, prefix: "pip-b-input-file");
+            FileArtifact bOutput = CreateOutputFileArtifact(root: dirPath, prefix: "pip-b-out-file");
+            File.WriteAllText(bInput.Path.ToString(Context.PathTable), "pipBOrigin");
+
+            var pipBuilderB = CreatePipBuilder(new[] { Operation.ReadFile(bInput), Operation.WriteFile(bOutput) });
+            var pipB = SchedulePipBuilder(pipBuilderB);
+
+            // Process C.
+            FileArtifact cOutput = CreateOutputFileArtifact(root: dirPath, prefix: "pip-c-out-file");
+            AbsolutePath changeAffectedWrittenFile = CreateUniqueObjPath("change");
+            var pipBuilderC = CreatePipBuilder(new[]
+            { 
+                // Ensure that pip reads the changeAffectedWrittenFile.
+                Operation.ReadFile(FileArtifact.CreateSourceFile(changeAffectedWrittenFile), doNotInfer: true),
+                Operation.ReadFile(aOutput),
+                Operation.ReadFile(bOutput),
+                Operation.WriteFile(cOutput)
+            });
+            pipBuilderC.SetChangeAffectedInputListWrittenFile(changeAffectedWrittenFile);
+            var pipC = SchedulePipBuilder(pipBuilderC);
+            Configuration.Schedule.InputChanges = changeList.Path;
+
+            // Build1 with change in aInput

```

Comment:
>// Build1 with change in aInput [](start = 12, length = 31)

Run Build0 where you don't have any change.

Reply:
Added